go/cmd/ensureconda.go

```diff
@@ -0,0 +1,229 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"github.com/Wessie/appdirs"
+	"github.com/hashicorp/go-version"
+	"github.com/spf13/cobra"
+	"os"
+	"runtime"
+	"strconv"
+)
+
+var (
+	// Used for flags.
+
+	rootCmd = &cobra.Command{
+		Use:   "ensureconda",
+		Short: "",
+		Long:  ``,
+		Run: func(cmd *cobra.Command, args []string) {
+			mamba, err := evaluateFlagPair(cmd, "mamba")
+			if err != nil {
+				panic(err)
+			}
+			micromamba, err := evaluateFlagPair(cmd, "micromamba")
+			if err != nil {
+				panic(err)
+			}
+			conda, err := evaluateFlagPair(cmd, "conda")
+			if err != nil {
+				panic(err)
+			}
+			condaExe, err := evaluateFlagPair(cmd, "conda-exe")
+			if err != nil {
+				panic(err)
+			}
+			noInstall, err := cmd.Flags().GetBool("no-install")
+			if err != nil {
+				panic(err)
+			}
+
+			executable, err := EnsureConda(mamba, micromamba, conda, condaExe, true)
```

Comment:
I understand this is out of scope - do we want to be able to optionally upgrade?